### PR TITLE
[Libp2p] - Sign Proposal Request Messages

### LINKED ISSUE #3602: [Libp2p] - Sign Proposal Request Messages
### What is this task and why do we need to work on it?

We need to sign PushCDN/Libp2p proposal request messages that are sent to the leader. This is based on the feedback from [this](https://github.com/EspressoSystems/HotShot/pull/3575) pull request.

### What work will need to be done to complete this task?

_No response_

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

Messages are signed and cannot be forged.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by jparr721**
Closes #3602 

### This PR: 
Implements message signing (again) for proposal request messages by signing the payload and computing a signature over the commitment.

### CODE REVIEW

File: crates/task-impls/src/helpers.rs [hunk 1]
```diff
@@ -84,9 +100,14 @@ pub(crate) async fn fetch_proposal<TYPES: NodeType, V: Versions>(
...
             if let Some(hs_event) = event.as_ref() {
-                if let HotShotEvent::QuorumProposalResponseRecv(_view_number, quorum_proposal) =
+                if let HotShotEvent::QuorumProposalResponseRecv(quorum_proposal) =
                     hs_event.as_ref()
                 {
+                    // Make sure that the quorum_proposal is valid
+                    if quorum_proposal.validate_signature(&mem).is_err() {
+                        return None
+                    }
+
```
Comment (bfish713) [inline crates/task-impls/src/helpers.rs:110]: Maybe we should just keep waiting or else we could just be attacked by giving bad proposal responses from anyone and end up ignoring the real response

**Review by bfish713 — COMMENTED** [review]
Just one small comment, I'll leave approval to the assigned reviewers

**Review by QuentinI — APPROVED** [review]
LGTM

File: crates/task-impls/src/helpers.rs (hunk 1 above)
Comment (jparr721) [reply crates/task-impls/src/helpers.rs]: Done

**Review by lukeiannucci — APPROVED** [review]
Thanks for changing the names, looks good.

**Pull request merged**